packages/ui/form/PhoneInput.tsx

```diff
@@ -18,17 +18,34 @@ export type PhoneInputProps = {
 };
 
 function BasePhoneInput({ name, className = "", onChange, ...rest }: PhoneInputProps) {
+  useEffect(() => {
+    if (!rest.value) {
+      return;
+    }
+    const formattedValue = rest.value.trim().replace(/^\+?/, "+");
+    onChange(formattedValue);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  let countryCode;
+  if (rest.value) {
+    try {
+      countryCode = parsePhoneNumber(rest.value)?.country || "";
+    } catch {}
+  }
+
   const defaultCountry = useDefaultCountry();
   return (
     <PhoneInput
       {...rest}
-      country={defaultCountry}
+      country={rest.value && countryCode ? undefined : defaultCountry}
```

Comment:
Have to done this change to see its an valid country code in phone number other wise it was changing to US